src/main/java/com/example/petbuddybackend/controller/ChatController.java

```diff
@@ -26,28 +32,133 @@ public class ChatController {
     private final ChatService chatService;
 
 
+    @PreAuthorize("isAuthenticated()")
     @GetMapping("/{chatId}/messages")
     @Operation(
             summary = "Get page of chat messages",
             description =
                     "Retrieves a paginated list of chat messages for a chat room. " +
                     "The page is sorted by message timestamp in descending order (from newest to oldest)."
     )
-    @PreAuthorize("isAuthenticated()")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved chat messages"),
+            @ApiResponse(responseCode = "404", description = "Chat room not found"),
+            @ApiResponse(responseCode = "403", description = "User is not a participant in the chat room")
+    })
     public Page<ChatMessageDTO> getChatMessages(
             Principal principal,
             @PathVariable Long chatId,
             @Valid @ParameterObject PagingParams pagingParams,
-            @Parameter(
-                    description = "The time zone to adjust the message timestamps to. If not provided, server's default timezone will be used.",
-                    schema = @Schema(type = "string", example = "Europe/Warsaw, CET, +00:02, ...")
-            )
-            @RequestHeader(value = "${timezone.header-name}", required = false) String acceptTimeZone
+            @TimeZoneParameter @RequestHeader(value = "${header-name.timezone}", required = false) Optional<String>  acceptTimeZone
+    ) {
+        Pageable pageable = PagingUtils.createPageable(pagingParams);
+
+        return acceptTimeZone.isPresent() ?
+                chatService.getChatMessages(
+                        chatId,
+                        principal.getName(),
+                        pageable,
+                        TimeUtils.getOrSystemDefault(acceptTimeZone.get())
```

Comment:
a tutaj musimy sprawdzac czy Zone podany, skoro jesli nie to i tak zwracamy default?